src/v/redpanda/admin_server.cc

```diff
@@ -738,6 +740,27 @@ ss::future<> admin_server::throw_on_error(
             throw ss::httpd::server_error_exception(
               fmt::format("Unexpected tx_error error: {}", ec.message()));
         }
+    } else if (ec.category() == rpc::error_category()) {
+        switch (rpc::errc(ec.value())) {
+        case rpc::errc::success:
+            co_return;
+        case rpc::errc::disconnected_endpoint:
+            [[fallthrough]];
+        case rpc::errc::exponential_backoff:
+            throw ss::httpd::base_exception(
+              fmt::format("Not ready: {}", ec.message()),
+              ss::httpd::reply::status_type::service_unavailable);
+        case rpc::errc::client_request_timeout:
+            throw ss::httpd::base_exception(
+              fmt::format("Timeout: {}", ec.message()),
```

Comment:
nit: I guess this will result in a "Timeout: timeout" message which is tautological